### PR TITLE
Redirect plain http to https filter

### PR DESCRIPTION
**Pull request by AmazingDreams**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [ ] Have you [squashed your commits]? (Optional, but makes merge messages nicer.)
* [x] Have you added copyright headers to new files?
* [ ] Have you checked that both Scala and Java APIs are updated?
* [ ] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Helpful things

## Purpose

This filter redirects http to https based on the `X-Forwarded-Proto` header and `request.secure`

## Background Context

I found myself writing this filter over and over so I thought it might be nice to have it in a core module.

I am willing to write some documentation for the changes I made but thought I'd wait for some feedback first.

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @AmazingDreams,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by typesafehub-validator** [comment]
Hi @AmazingDreams,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: framework/src/play/src/main/scala/play/api/http/StandardValues.scala [hunk 1]
```diff
@@ -306,6 +306,8 @@ trait HeaderNames {
   val ORIGIN = "Origin"
   val ACCESS_CONTROL_REQUEST_METHOD = "Access-Control-Request-Method"
   val ACCESS_CONTROL_REQUEST_HEADERS = "Access-Control-Request-Headers"
+
+  val STRICT_TRANSPORT_SECURITY = "Strict-Tranpsort-Security"
```
Comment (mkurz) [inline framework/src/play/src/main/scala/play/api/http/StandardValues.scala]: typo

Comment (AmazingDreams) [inline framework/src/play/src/main/scala/play/api/http/StandardValues.scala]: Thanks, fixed

**Comment by PromanSEW** [comment]
Looks that Configuration is removing in PR #6823 
So you have to change using this class in future

**Comment by marcospereira** [comment]
Hey @AmazingDreams, 

What is being deprecated (and will be removed in the future) is the Java wrapper for `Config`, `play.Configuration`. `play.api.Configuration` is the Scala version of this API and it was not affected by #6823.

That means you are safe to use it.

**Comment by gmethvin** [comment]
I think the main reason this is not included in Play already is that you would typically do this in a reverse proxy. But I have no problem with including something like this in Play as long as it's well tested and documented.

**Comment by AmazingDreams** [comment]
@marcospereira Ok, so I guess I best revert that commit so we only use scala APIs?

@gmethvin Yes, I think that usually it is handled at the proxy level. I am currently running Docker containers on Kubernetes using a [generic proxy](https://github.com/kubernetes/contrib/tree/master/ingress/controllers/nginx) based on Ingresses (rules that specify which container to use based on the host and path). I run this on AWS and this generic proxy itself is behind an Elastic Loadbalancer.

Anyhow, this generic proxy doesn't redirect SSL because I haven't configured it with SSL. The ELB is though. I'm not comfortable writing Go let alone working on a kubernetes plugin. So I wrote this and figured I might just as well create a PR.

@gmethvin (2) Currently the code reads the `X_FORWARDED_PROTO` header and checks if it is http or https, if it cannot find it it falls back to `request.secure`. I'm not sure what you mean with it should parse the `Forwarded` header entry.

**Comment by AmazingDreams** [comment]
@typesafehub-validator I signed the contributor CLA

**Comment by gmethvin** [comment]
@AmazingDreams For identifying proxies Play currently supports both the `X-Forwarded-*` headers and the `Forwarded` header. The `Forwarded` header is the official standard defined in RFC7239. It contains a `proto` field that can be used to identify the protocol: https://tools.ietf.org/html/rfc7239#section-5.4

A forwarded header looks something like:
```
Forwarded: for=192.0.2.60;proto=http;by=203.0.113.43
```

We already do parsing of forwarded headers in the [ForwardedHeaderHandler](https://github.com/playframework/playframework/blob/master/framework/src/play-server/src/main/scala/play/core/server/common/ForwardedHeaderHandler.scala), so I was suggesting we could save more of that information on the request (perhaps on a request attribute, a new feature we're adding in 2.6) and use it here.

**Comment by gmethvin** [comment]
@AmazingDreams Also see the discussion on #6845.

**Comment by AmazingDreams** [comment]
@gmethvin it is more complex than I thought. I have included the Forwarded header in my checks for https, though I don't do any sanity checking.

Everything is in place to simply remove the checks on the headers and simply use `req.secure`. Which isn't really working with the proxies right now (in my tests at least).

**Comment by benmccann** [comment]
@AmazingDreams this PR needs a rebase

**Comment by wsargent** [comment]
Closing and using https://github.com/playframework/playframework/pull/7071 instead...

**Pull request closed without merging**